.github/workflows/commit.yaml

```diff
@@ -4,8 +4,11 @@
 ---
 name: "build"
 
-env:  # Update this prior to requiring a higher minor version in go.mod
-  GO_VERSION: "1.17"  # Latest patch
+# GOROOT_NAME resolves a GOROOT path corresponding to a major Golang release on the current runner.
+# Ex. Given GOROOT_NAME=GOROOT_1_17_X64, ${!GOROOT_NAME}=/opt/hostedtoolcache/go/1.17.1/x64
+# See https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#environment-variables-2
+env:
```

Comment:
It occurs to me that now that we use shell to setup go, we can use that to parse go.mod to get the correct version (as @mathetake suggested in a different PR). This means we choose the runtime conflated with the minimum go version, but IMHO that's ok. The other thing this means is we must checkout the source prior to setting up go, but there's only one case where we don't already do that.

Basically I think we can remove the need to set GOROOT_NAME=GOROOT_1_17_X64 as long as we are ok conflating the runtime version with the version declared in go.mod

Reply:
making a general PR comment about this problem and a way out